addons/web/static/src/core/popover/popover.js

```diff
@@ -11,7 +11,21 @@ export class Popover extends Component {
             position: this.props.position,
         });
     }
-    onPositioned(el, { direction, variant }) {
+    onPositioned(el, { top, left, direction, variant }) {
+        const arrowEl = el.firstElementChild;
+        
+        // Modify positioning if the popup has an arrow
+        const arrowBox = arrowEl.style.display != "none" && arrowEl.getBoundingClientRect();
+        if (arrowBox) {
+            if (direction === "left") {
+                left -= arrowBox.height;
+                el.style.left = `${left}px`;
+            } else if (direction === "top") {
+                top -= arrowBox.height;
+                el.style.top = `${top}px`;
+            }
+        }
+        
```

Comment:
```suggestion
```

Instead do in popover.scss
```patch
diff --git a/addons/web/static/src/core/popover/popover.scss b/addons/web/static/src/core/popover/popover.scss
index e9b38e8408b2..8bf161844648 100644
--- a/addons/web/static/src/core/popover/popover.scss
+++ b/addons/web/static/src/core/popover/popover.scss
@@ -36,7 +36,7 @@
 	}
 
 	&.o-popover-top, &.o-popover-auto[x-placement^="top"] {
-		margin-bottom: 0.5rem;
+		margin-top: -0.5rem;
 	}
 
 	&.o-popover-right, &.o-popover-auto[x-placement^="right"] {
@@ -48,7 +48,7 @@
 	}
 
 	&.o-popover-left, &.o-popover-auto[x-placement^="left"] {
-		margin-right: 0.5rem;
+		margin-left: -0.5rem;
 	}
 
 	&.o-popover--ts .popover-arrow, &.o-popover--bs .popover-arrow {
```

Reply:
> In fact I have a better fix:
> 
> * replace
> 
> ```css
> 	&.o-popover-top, &.o-popover-auto[x-placement^="top"] {
> 		margin-bottom: 0.5rem;
> 	}
> 
> 	&.o-popover-right, &.o-popover-auto[x-placement^="right"] {
> 		margin-left: 0.5rem;
> 	}
> 
> 	&.o-popover-bottom, &.o-popover-auto[x-placement^="bottom"] {
> 		margin-top: 0.5rem;
> 	}
> 
> 	&.o-popover-left, &.o-popover-auto[x-placement^="left"] {
> 		margin-right: 0.5rem;
> 	}
> ```
> 
> * with
> 
> ```css
> 	margin: 0.5rem;
> ```
> 
> I gave it a try in 16.0 and in master branches.
> 
> You'll need to make it conditional in master though, as the arrow can now be disabled.

Thank you @brboi for the feedback and sorry for the delay to make the changes. Your last suggestion mess up with the arrow when it is displayed (e.g. for the activity popup in saas-16.3, see the task description here: https://www.odoo.com/web#id=3457106&cids=1&menu_id=4720&action=333&active_id=4105&model=project.task&view_type=form)

The fact that we add left margin when it is not needed (when the popup is positioned above/below an element) mess the alignment of the arrow. So What I've done here
- Remove my changes in javascript and use your penultimate suggestion
- Remove the test as the change does only affect style now
- Re-target the pr to 16.0

Let me know if you agree with that.